src/extension.ts

```diff
@@ -451,20 +451,27 @@ class ContextDecorator {
  */
 class Completion {
 	static readonly runningCompletions = new Set<Completion>();
+	private textEditor: vscode.TextEditor;
 	private document: vscode.TextDocument;
 	private anchorOffset: number;
 	private changes: Set<string>;
 	private abortController?: AbortController;
+	private completionIndicator: vscode.TextEditorDecorationType;
 
-	constructor(document: vscode.TextDocument, anchorOffset: number) {
-		this.document = document;
+	constructor(textEditor: vscode.TextEditor, anchorOffset: number) {
```

Comment:
Changing the constructor argument from `TextDocument` to `TextEditor` is a good idea 👍